tests/src/test/scala/com/evolutiongaming/skafka/ProducerConsumerSpec.scala

```diff
@@ -29,15 +29,18 @@ import scala.concurrent.duration._
 class ProducerConsumerSpec extends AnyFunSuite with BeforeAndAfterAll with Matchers {
   import ProducerConsumerSpec._
 
-  private lazy val shutdown = StartKafka()
+  private val kafkaContainer = KafkaContainer()
+
+  // We need to start the container before the test suite, because the container might not be ready yet when the
+  // `consumerOf` / `producerOf` method is called, and the class's `init` method is called before the `beforeAll` method
+  kafkaContainer.start()
```

Comment:
could we use some better mechanism of ensuring broker is launched? periodic pinging?

Reply:
Not without rewriting the structure of the test, it seems. If `kafkaContainer.bootstrapServers` is called before `start`, it's going to fail with an error immediately, and we need `bootstrapServers` in order to complete the class' `init` method (executing `consumerOf`/`producerOf`). Execution is happening like that: class's `init` method -> test's `beforeAll` method -> actual test, and `bootstrapServers` is called before producer/consumer needs it